### PR TITLE
MRG: hrf refactor

### PR DESCRIPTION
**Pull request by bthirion**
This is an update of hrf module. Fixes a numerical issue, improves docstrings, adds  an example.
Addresses #36 and #34. 
The most debattable change is the removal of the name 'canonical' for the 'glover' hrf, because this name is unclear (in spm, the canonical hrf is not that one). This will break the compatibility with nipy.
What do people think about that ?

### CODE REVIEW

**Comment by MartinPerez** [comment]
Why not keep the canonical (the one implemented in spm) and then also add the glover in the documentation and examples as an option? Someone like working with fmri experiments would like to see a link to some paper or something that justifies the deviation from the SPM standards that are hardwired in the community. I can imagine a reviewer asking for the SPM canonical results alongside the glover one.

**Comment by bthirion** [comment]
I understand your point, but so far, what was called the 'canonical' hrf so far was the 'Glover' one !
This means that the term "canonical" is actually not so canonical...
So I prefer not to use it any more. Taking the spm hrf as a default feels right. 
I can change my mind if people believe that 'canonical' should be used instead (talking about the 'spm' hrf).
ping @agramfort ?

**Comment by agramfort** [comment]
canonical for spm or for freesurfer does not mean the same thing.

I would avoid canonical. It's not specific.

**Comment by MartinPerez** [comment]
all here seems fine to me

**Comment by bthirion** [comment]
No further comment on that one ? Note that it breaks back-ward compatibility with nipy...

File: nistats/design_matrix.py [hunk 1]
```diff
@@ -271,9 +276,9 @@ def make_design_matrix(
     paradigm : DataFrame instance, optional
         Description of the experimental paradigm.
 
-    hrf_model : string, optional,
-        Specifies the hemodynamic response function (HRF).
-        It can be 'canonical', 'canonical with derivative' or 'fir'
+    hrf_model : {'spm', 'spm + derivative', 'spm + derivative + dispersion',
+        'glover', 'glover + derivative', 'fir'}, optional,
```
Comment (MartinPerez) [inline nistats/design_matrix.py]: 'glover + derivative + dispersion' missing?

File: nistats/hemodynamic_models.py [hunk 2]
```diff
@@ -327,15 +329,15 @@ def _regressor_names(con_name, hrf_model, fir_delays=None):
     names: list of strings,
         regressor names
     """
-    if hrf_model == 'canonical':
+    if hrf_model == 'glover':
         return [con_name]
-    elif hrf_model == "canonical with derivative":
+    elif hrf_model == "glover + derivative":
         return [con_name, con_name + "_derivative"]
```
Comment (MartinPerez) [inline nistats/hemodynamic_models.py]: 'glover + derivative + dispersion' missing?

File: nistats/tests/test_dmtx.py [hunk 3]
```diff
@@ -176,7 +176,7 @@ def test_design_matrix1b():
     tr = 1.0
     frame_times = np.linspace(0, 127 * tr, 128)
     paradigm = basic_paradigm()
-    hrf_model = 'Canonical'
+    hrf_model = 'Glover'
```
Comment (MartinPerez) [inline nistats/tests/test_dmtx.py]: 'Glover' with uppercaps instead of 'glover' like everywhere else.

**Comment by MartinPerez** [comment]
Just one question, any reason the glover has no dispersion option?

**Comment by bthirion** [comment]
I think I have addressed your comments. Thx.

**Comment by MartinPerez** [comment]
@bthirion can we merge this? to avoid having inconsistent hrf naming in coming api changes and examples.

**Pull request merged**